src/_pytest/config/__init__.py

```diff
@@ -1351,6 +1356,27 @@ def getvalueorskip(self, name: str, path=None):
         """ (deprecated, use getoption(skip=True)) """
         return self.getoption(name, skip=True)
 
+    def _warn_about_missing_assertion(self, mode: str) -> None:
+        if not _assertion_supported():
+            from _pytest.warnings import _issue_warning_captured
+
+            warning_text = (
```

Comment:
I somewhat prefer the way it was written before (if/else), seems more symmetric to me.

And potentially we might use `Literal` and `_pytest.compat.assert_never` on `mode`.

Or maybe even [`match` statements](https://www.python.org/dev/peps/pep-0622/) 10 years from now :)

Reply:
I agree. 👍

Reply:
changed it back to if/else